src/Utilities/RandomGenerator.h

```diff
@@ -38,29 +38,18 @@ uint32_t make_seed(int i, int n);
 
 namespace qmcplusplus
 {
-template<class RNG>
-class RNGThreadSafe : public RNG
-{
-public:
-  using result_type = typename RNG::result_type;
-
-  /** return a random number [0,1)
-   */
-  result_type operator()();
-};
-
-extern template class RNGThreadSafe<FakeRandom>;
-extern template class RNGThreadSafe<StdRandom<double>>;
+extern template class FakeRandom<double>;
+extern template class StdRandom<double>;
 
 #if defined(USE_FAKE_RNG)
 // fake RNG redirection
-using RandomGenerator = FakeRandom;
-extern RNGThreadSafe<RandomGenerator> fake_random_global;
+using RandomGenerator = FakeRandom<OHMMS_PRECISION_FULL>;
+extern RandomGenerator fake_random_global;
 #define Random fake_random_global
 #else
 // real RNG redirection
 using RandomGenerator = StdRandom<OHMMS_PRECISION_FULL>;
-extern RNGThreadSafe<RandomGenerator> random_global;
+extern RandomGenerator random_global;
```

Comment:
Is it safe to drop the RNGThreadSafe wrapper?

Reply:
I moved the wrapper into `StdRandom::operator()()` and `FakeRandom::operator()()`

Reply:
In the old code, critical is only needed on the random_global. Now the cost is paid on very call to generate a RN.

Reply:
Then is it safe to have that critical section used everywhere?  It looks like previously the wrapper is only used to lock the global RNG instance.

Reply:
Yep, that change looks like a mistake. I'll revert it.

Reply:
The `RNGThreadSafe` wrapper should work as before.